upstream merge

kaue opened the pull request:
would be nice to have some of those bugfixes merged into this fork.
@josteink can you check if there is anything useful?
> 23 commits behind the upstream and has conflicts that must be resolved.

YorkZ committed:
Always display inferior process buffer after evaluation

Previously, if the inferior process buffer is not being shown in any
window, one would have to manually bring it up after evaluating an
expression in order to see the results. With this change, the inferior
process buffer will always be displayed after evaluating an expression
so that one can alway see the results without having to bring up the
inferior process buffer.

redguardtoo committed:
Merge pull request #11 from YorkZ/york

Always display inferior process buffer after evaluation

redguardtoo committed:
add js-clear

redguardtoo committed:
ignore shebang (first line) of node js script

redguardtoo committed:
could hide `undefined` output

redguardtoo committed:
could setup $NODE_PATH automatically

redguardtoo committed:
clean code and README

redguardtoo committed:
fixed typo

redguardtoo committed:
Learn from nodejs-repl

redguardtoo committed:
use js-comint-module-paths to search for node module

redguardtoo committed:
fixed typo

redguardtoo committed:
fixed bugs. evil-mode friendly

redguardtoo committed:
upgrade license

redguardtoo committed:
doc and bugs

Commit:
Update docs

redguardtoo committed:
Merge pull request #22 from re-mx/update-docs

Update docs

Commit:
Fixed typo

I am very sorry... I just typed a typo...

redguardtoo committed:
Merge pull request #23 from re-mx/patch-1

Fixed typo

josteink committed:
Add GPL3 license

This closes https://github.com/redguardtoo/js-comint/issues/24.

redguardtoo committed:
Merge pull request #25 from josteink/master

Add GPL3 license

Commit:
defining our adjust-window-size-function

redguardtoo committed:
Merge pull request #26 from saadnpq/fix-dub-prompts

defining our adjust-window-size-function

redguardtoo committed:
add node_modules into repl automatically

josteink commented:
Seems like quite a bit of things have diverged since I made my small fork, and that merging won't be trivial.

I'm definitely not against trying to better align with a new upstream version, but I'm not sure I will have time to follow this closely any day soon.

If you could help try to resolve some of the merge-conflicts, I could probably more easily review this PR and see if I think the changes makes sense for this project too :)

kaue commented:
Indeed this seems like a quite difficult merge to make. I may try to help with a couple of merges, but i don't think have the required knowledge in this code base / elisp for this task.

But yeah, this would be a great effort so we can continue merging with the upstream getting new features / bugfixes :)

DivineDominion committed:
Update key bindings

redguardtoo committed:
Merge pull request #29 from DivineDominion/patch-1

Update key bindings

josteink commented:
GitHub says there’s conflicts which needs to be resolved. Are you looking into that? 🙂

linktohack committed:
Allow to send multi-line string

redguardtoo committed:
Merge pull request #32 from linktohack/patch-2

Allow to send multi-line string

saffroy committed:
Fixes to run under Emacs 27.1

redguardtoo committed:
Merge pull request #33 from saffroy/master

Fixes to run under Emacs 27.1

saffroy committed:
Fix regression when sending multi-line string

redguardtoo committed:
Merge pull request #34 from saffroy/master

Fix regression when sending multi-line string

josteink commented:
I really appreciate the endurance here. This is the PR which comes back from the dead and just refuses to die 😄 

<img width="919" alt="image" src="https://github.com/emacs-typescript/ts-comint/assets/411338/b2e77abe-2202-43fa-b0a1-dd474e54f7b3">

That said, like I've already mentioned, there are merge conflicts and I cannot merge this PR without those getting resolved.

Can you take a look?

saffroy committed:
Add tests using Emacs ERT for multiline JS statements

saffroy committed:
Add github workflow that runs tests across multiple Emacs versions

redguardtoo committed:
Merge pull request #35 from saffroy/master

Add tests for multiline statements, run tests in github workflow

josteink commented:
@kaue & @saffroy :  Nice update, but you are still not resolving the conflicts, so I cannot merge this....

saffroy commented:
@josteink I don't quite understand what you expect from me here.

I sent PRs to the upstream project, they were merged there without conflict.

Anything that happens in a fork from upstream is not something I can be aware of (and I wasn't until you pinged me here), nor is it something I should be concerned with simply because it exists.

It looks like this PR automatically tracks the master branch from upstream, so of course it's a moving target, and upstream changes can become conflicts here. Maybe that was a mistake?

joshbax189 committed:
Enable lexical binding

joshbax189 committed:
Add lexical-binding and requires to test

joshbax189 committed:
Ignore string arg to js-comint-filter-output

It just changes the output in the comint buffer.

joshbax189 committed:
Lint and simplify js-comint-repl

require does not load if already present.
Convert cond to equivalent if

joshbax189 committed:
Bug: js-comint--suggest-module-path should be nil if no node_modules

This means that default-directory is always added to NODE_PATH
when calling js-comint-repl

joshbax189 committed:
Use temporary local env vars when calling comint

This avoids modifying NODE_PATH for later calls

joshbax189 committed:
Remove unused js-comint-setup-module-paths

joshbax189 committed:
Depend on emacs 28.1 to use with-environment-variables

joshbax189 committed:
Tidy argument handling for js-comint-repl

Made cmd optional and documented behaviour.

joshbax189 committed:
Add missing customization types to fix compiler warnings

joshbax189 committed:
Add unit tests for js-comint-start-or-switch-to-repl

joshbax189 committed:
Simplify js-comint--suggest-module-path

read-directory-name defaults to default-directory if nil
and expand-file-name produces output equivalent to file-truename.

joshbax189 committed:
Have js-comint-start-or-switch-to-repl check local node_modules

This way js-rest-repl does not "forget" the local node_modules.

joshbax189 committed:
Fix failing test

joshbax189 committed:
Factor out process exit logic in tests

joshbax189 committed:
Add declares for nvm functions.

This fixes byte-compiler warnings, but keeps nvm optional

joshbax189 committed:
Move (require 'nvm) into select-node-version, tidy logic

js-comint-select-node-version is the common entry point when using
nvm functions.

joshbax189 committed:
Message when changing cmd while nvm is set

joshbax189 committed:
Do not include empty strings in path

joshbax189 committed:
Switch buffers in start-or-switch-to-buffer

redguardtoo committed:
Merge pull request #37 from joshbax189/feat/lexical-binding

Feat/lexical binding

joshbax189 committed:
Add Eask file

Add gitignore

joshbax189 committed:
Update github action to use eask

Remove unused script files

joshbax189 committed:
repl.start() takes a single object arg

joshbax189 committed:
NODE_REPL_MODE does not need to be set manually

joshbax189 committed:
Add lint job to catch nvm.el signature changes

joshbax189 committed:
Add unit tests for js-comint-select-node-version using el-mock

redguardtoo committed:
Merge pull request #38 from joshbax189/fix/update-node-args

Update args for repl.start()

redguardtoo committed:
Merge pull request #39 from joshbax189/feat/use-eask

Feat: use eask package manager

jian-lin committed:
Exclude tests from MELPA packages

MELPA excludes[1] files named as *-test.el.

[1]: https://github.com/melpa/melpa/blob/f1945a7004f8bbe4043a8579b16e1147f19d7623/README.md?plain=1#L171

redguardtoo committed:
Merge pull request #40 from linj-fork/pr/exclude-test-files-from-melpa

Exclude tests from MELPA packages

joshbax189 committed:
Add completion feature

Squashed commit of the following:

commit 479eb4664d9f7325980bd61e344bd7e631aa57fd
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Dec 3 16:41:38 2024 -0800

    Fix: use emacs 28 compat completion fns in test

commit 5870d05b9417e3a46f3806cc76089a051842fe6a
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Dec 3 16:17:33 2024 -0800

    Fix: avoid Eask issue to unblock tests

commit 19446fd9f146e92500ed511ddd57d42219091f8f
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Dec 3 10:51:17 2024 -0800

    Tidy comments and docstrings

commit af87be00c890c506b562eee473d97d71eca1cae8
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Dec 3 10:41:40 2024 -0800

    Fix do not pass arguments to callback

commit ee0824b7ebeb3f7bef8830e368de3b670c2e34fa
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Dec 3 15:23:16 2024 -0800

    Add Capf tests

    * test/capf-tests.el: C

commit 9fcb825af2fe384b822361d1d15ca6f142c7a627
Author: Josh Bax <joshbax189@gmail.com>
Date:   Mon Dec 2 17:50:10 2024 -0800

    Move tests into own folder and separate files

    This allows separate loading of company and CAPE

commit bd2616b48575fcb66fb5e438b273f23486ced759
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Dec 3 15:17:17 2024 -0800

    Native capf

commit 5d67b6ba8594214ec963ef5f93cd94892eb8e6c7
Author: Josh Bax <joshbax189@gmail.com>
Date:   Wed Nov 27 11:22:20 2024 -0800

    Use string-match-p to support 28

commit 9f1241f22e0eee9f3cbdf52ea391ceee2696b33e
Author: Josh Bax <joshbax189@gmail.com>
Date:   Wed Nov 27 11:03:56 2024 -0800

    Cleanup hung callbacks when completing props of null object

commit 89375779aa38aae4990267187bd54243fd35755f
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Nov 26 12:45:12 2024 -0800

    Quote string when used as regexp in callback

commit e276f26555b458790d865205b8ec88fd86de4433
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Nov 26 11:46:38 2024 -0800

    Address fast-typing bug: extra input when callbacks overlap

commit 4353c89f0e263f664c09d531dab1e38ba30aa6f5
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Nov 26 10:28:00 2024 -0800

    Use ert-async for tests with checks in callbacks

    Update test-company-complete-long-line

commit 44f2eacd1e030d372ef5dfa3c9e1e36e03bec98b
Author: Josh Bax <joshbax189@gmail.com>
Date:   Mon Nov 25 12:32:37 2024 -0800

    Fix matching bug in complete-substring

commit bb479f5f3663a33a59fea97e91bcfcc9e8302bf6
Author: Josh Bax <joshbax189@gmail.com>
Date:   Mon Nov 25 11:34:27 2024 -0800

    Remove unused completion string property

commit ea7a9fd68810059b4e9e0b51a5885dad8bf23d62
Author: Josh Bax <joshbax189@gmail.com>
Date:   Mon Nov 25 10:28:29 2024 -0800

    Fix: minor formatting change

commit 2ae491d162e49b6109ec74351f432fa50a873e22
Author: Josh Bax <joshbax189@gmail.com>
Date:   Mon Oct 28 22:35:16 2024 -0700

    Cleanup completion buffers

commit e07de7bb02114529b5abf962a50431b70002fa80
Author: Josh Bax <joshbax189@gmail.com>
Date:   Mon Oct 28 22:22:11 2024 -0700

    Pass NODE_REPL_MODE as argument due to custom env

    Env var has no effect for some reason

commit b649559fb0055a29b5e0b093b0420a21bd618608
Author: Josh Bax <joshbax189@gmail.com>
Date:   Mon Oct 28 22:19:17 2024 -0700

    Wiggle the cursor to produce a prompt and prevent hanging

    This makes most cases resolve with similar output too

commit 123c87dec4434a9a6e3aa87f2b26306367325c12
Author: Josh Bax <joshbax189@gmail.com>
Date:   Mon Oct 28 21:23:52 2024 -0700

    Add Company integration tests

commit 83f8a31b31fbb63780618ce48f20bc81eb79cf6f
Author: Josh Bax <joshbax189@gmail.com>
Date:   Mon Oct 28 19:37:40 2024 -0700

    Simplify process-completion-output

commit f80ca422a6991690489554a710128fa99c0016d1
Author: Josh Bax <joshbax189@gmail.com>
Date:   Mon Oct 28 19:24:50 2024 -0700

    Proactively clear partial completions

commit 672c28afe896cf8cc60f1c641dbc7b8cae9465e5
Author: Josh Bax <joshbax189@gmail.com>
Date:   Mon Oct 28 16:08:57 2024 -0700

    Complete long statements by taking latest substring

commit 3e6863ef2f0f9064bdbdae72ad349416a4c984ee
Author: Josh Bax <joshbax189@gmail.com>
Date:   Mon Oct 28 15:48:04 2024 -0700

    Don't complete at whitespace

commit 51ebba7051e852e1e3e7d36e0bdc92d81fa3fd42
Author: Josh Bax <joshbax189@gmail.com>
Date:   Fri Oct 25 14:48:24 2024 -0700

    Add unit tests for listener callback code

commit 41577969df45572ef4e6d7cc1b9e9a734f117f57
Author: Josh Bax <joshbax189@gmail.com>
Date:   Fri Oct 25 14:42:12 2024 -0700

    Recover from errors in company callbacks after completion

commit 0ed6d0be08ef334c710a2dcd0ab67d5d95e91098
Author: Josh Bax <joshbax189@gmail.com>
Date:   Fri Oct 25 14:12:50 2024 -0700

    Set TERM using env

    Remove with-environment-variables because it must be set within the
    comint call. Move other vars into the same location.

commit aa3821050361f608469924f6144b33e2f186244c
Author: Josh Bax <joshbax189@gmail.com>
Date:   Fri Oct 25 10:22:44 2024 -0700

    Failed callbacks should be removed

commit 5b3efeaf6bde14e3729e80bea24bfc48e67a367e
Author: Josh Bax <joshbax189@gmail.com>
Date:   Fri Oct 25 09:45:53 2024 -0700

    Factor out callback active test

commit 67208a6cf769d4250342d46e61b6298999f9dbd4
Author: Josh Bax <joshbax189@gmail.com>
Date:   Thu Oct 24 23:11:50 2024 -0700

    Add retry counter for method completion

commit 9433d55aea03aeea9a9167451425d7a1e4bbcc42
Author: Josh Bax <joshbax189@gmail.com>
Date:   Thu Oct 24 23:11:38 2024 -0700

    Factor out callback creation

commit dd0a5e5deac9608c692f356b7be9767df2650adc
Author: Josh Bax <joshbax189@gmail.com>
Date:   Thu Oct 24 13:08:07 2024 -0700

    Rewrite completions to use listener pattern callback

commit 83e3509a5d472d3b7b90c42daf394eb218a20813
Author: Josh Bax <joshbax189@gmail.com>
Date:   Thu Oct 24 13:02:45 2024 -0700

    Rename company-js-comint-backend => company-js-comint

    Matches other backend names

commit 74592d57755a51e88508f41707cc636bf09a475a
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Oct 22 23:56:41 2024 -0700

    Don't use comint-send-string for invisible input

commit 4816ab3dce13bfc40fcf1dc0a880b883d9c247a0
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Oct 22 23:50:09 2024 -0700

    Inline company definition in should-complete

commit c6cbbda5b43028aa6aa995b736dcda93429494ee
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Oct 22 12:48:22 2024 -0700

    Use 28.1 compatible kill-matching-buffers in test

commit bb64f539258354b2365a965470a9ca1475e79749
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Oct 22 12:36:52 2024 -0700

    Add should-complete unit tests

    Add Eask file

commit 0dc206a7be64f61ba8098f22a962be317d115df6
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Oct 22 11:23:47 2024 -0700

    Add test macro with-new-js-comint-buffer and simplify tests

    Kills comint buffer between runs for safety

commit 054c09781d9c2c1711b4e56b822650cb44169944
Author: Josh Bax <joshbax189@gmail.com>
Date:   Sun Oct 20 11:42:37 2024 -0700

    Add unload function to remove company backend

commit 96cf5a7d7f45402d67523e52872c5ffed9099f9c
Author: Josh Bax <joshbax189@gmail.com>
Date:   Wed Oct 16 16:32:28 2024 -0700

    Unit tests for normal completion

commit 2721fd884b06eeb06e456bfb78a9c52c5c33d704
Author: Josh Bax <joshbax189@gmail.com>
Date:   Wed Oct 16 16:24:30 2024 -0700

    Use js-comint-get-process

commit f2f3d65c8729acbf585c3494ab0768d526161276
Author: Josh Bax <joshbax189@gmail.com>
Date:   Wed Oct 16 15:50:56 2024 -0700

    Do not take arguments for js-comint--discard-output when a callback

commit 7829facfcbf47b90714391df3bc5748ead32e438
Author: Josh Bax <joshbax189@gmail.com>
Date:   Wed Oct 16 15:41:30 2024 -0700

    Discard should apply to all output until control char

commit 6e274c8fe3c50247a6481b1d6b943e1fd1b8bbc6
Author: Josh Bax <joshbax189@gmail.com>
Date:   Wed Oct 16 15:35:53 2024 -0700

    Add unit tests for js-comint--process-completion-output

commit 5b50adad45d65b73b5480b8d0b589b965cac068c
Author: Josh Bax <joshbax189@gmail.com>
Date:   Wed Oct 16 13:32:57 2024 -0700

    Properly complete method names

commit 7f57d7618e43ed43e1d7a47b46cac22ed4825b59
Author: Josh Bax <joshbax189@gmail.com>
Date:   Wed Oct 16 11:12:29 2024 -0700

    Factor out completion processing logic

commit a98dc6e4947a085eacc82f68539ad72638e3f4a3
Author: Josh Bax <joshbax189@gmail.com>
Date:   Wed Oct 16 09:15:20 2024 -0700

    Simplify handler logic

commit f182a523725c6e62a5c1808c2141c191d7c31acf
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Oct 15 15:30:25 2024 -0700

    Completion handler with debug messages

commit 18ff4f64a0426885b6e79cb6a130284a2082cdde
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Oct 15 13:43:59 2024 -0700

    Filter completion output when a flag is set

commit 1cfedc9854d4ce6f99a3a14132764ef720737955
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Oct 15 10:13:52 2024 -0700

    Get input for completion

commit 1352dbf197c4cc27ffdd82e57e01cc02171bf367
Author: Josh Bax <joshbax189@gmail.com>
Date:   Tue Oct 15 09:01:49 2024 -0700

    Add stub for company-js-comint-backend and init code

redguardtoo committed:
Merge pull request #41 from joshbax189/feat/completion

Input completion

redguardtoo committed:
support emacs 27

redguardtoo committed:
Revert "support emacs 27"

This reverts commit 1d2c27d3a6a333f47f2b34b0d61ff234e305bdca.
